java/ql/test/utils/modelgenerator/dataflow/p/Sinks.java

```diff
@@ -38,4 +44,24 @@ public void propagate(String s) {
     Logger logger = Logger.getLogger(Sinks.class.getSimpleName());
     logger.warning(s);
   }
+
+  // New sink as the value of a public field is propagated to a sink.
+  // sink=p;Sinks;true;fieldSink;();;Argument[this];test-sink;df-generated
+  // neutral=p;Sinks;fieldSink;();summary;df-generated
+  public void fieldSink() {
+    sink(tainted);
+  }
+
+  // Not a new sink as this method is already defined as a manual
+  // sink neutral.
+  // neutral=p;Sinks;hasManualSinkNeutral;(Object);summary;df-generated
+  public void hasManualSinkNeutral(Object o) {
+    sink(o);
+  }
+
+  // sink=p;Sinks;true;compoundPropgate;(Sinks);;Argument[0];test-sink;df-generated
+  // neutral=p;Sinks;compoundPropgate;(Sinks);summary;df-generated
```

Comment:
Why is it both neutral and a sink?

Reply:
Because it is "neutral" with respect to "summary" (and not "sink").